### PR TITLE
don't add binstar token if it's already in the url

### PR DESCRIPTION
**Pull request by kalefranz**
CC @dsludwig

### CODE REVIEW

File: conda/connection.py [hunk 1]
```diff
@@ -107,8 +108,11 @@ def get_binstar_token(url):
             return None
 
     @staticmethod
-    def is_binstar_url(url):
+    def is_binstar_url_needing_token(url):
         urlparts = urlparse(url)
+        token, path = split_token(urlparts.path)
```
Comment (pzwang) [inline conda/connection.py]: Can split_token() ever fail?  Is there a situation where it doesn't fail, but returns an invalid/empty token?

File: conda/connection.py [hunk 2]
```diff
@@ -107,8 +108,11 @@ def get_binstar_token(url):
...
+        if token:  # url already has token
+            return False
         return (urlparts.scheme in ('http', 'https') and
                 any(urlparts.hostname.endswith(bh) for bh in context.binstar_hosts))
```
Comment (pzwang) [inline conda/connection.py]: Why `hostname.endswith` and not an exact (case-insensitive) match?  Are we making an assumption here about the nature/structure of hostnames for the server, and if so, should we explicitly document that?  (e.g. that it's impossible to exclude a sub-domain `foo.bar.com` if `bar.com` is in the list of `binstar_hosts`?)

File: conda/connection.py (hunk 1 above)
Comment (kalefranz) [inline conda/connection.py]: `split_token` returns `None` if no token part of the path exists.

File: conda/connection.py (hunk 2 above)
Comment (kalefranz) [inline conda/connection.py]: Yeah this still isn't exactly correct.  I spent some time with @dsludwig yesterday trying to get all of this interaction with anaconda-client right, but we need more.  Going to schedule some time for next week.

Comment (kalefranz) [inline conda/connection.py]: > should we explicitly document that

Yes, in general I could have used more documentation/specification about intended behavior.  For example, just learned that if we use bare IP addresses for hostname, we use `/api` and `/conda` as part of the path.

**Pull request merged**